ViAn/GUI/mainwindow.cpp

```diff
@@ -389,3 +396,88 @@ bool MainWindow::eventFilter(QObject *obj, QEvent *event) {
     }
     return false;
 }
+/**
+ * @brief MainWindow::prepare_menu
+ * @param pos
+ * Creates context menu on right-click in tree view
+ */
+void MainWindow::prepare_menu(const QPoint & pos) {
+    QTreeWidget *tree = ui->ProjectTree;
+
+    MyQTreeWidgetItem *item = (MyQTreeWidgetItem*)tree->itemAt( pos );
+
+    QMenu menu(this);
+
+    if(item == nullptr) {
+
+    } else if(item->type == TYPE::PROJECT) {
+        selectedProject = item;
+        QAction *addVideo = new QAction(QIcon(""), tr("&Add video"), this);
+        addVideo->setStatusTip(tr("Add video"));
+        menu.addAction(addVideo);
+        connect(addVideo, SIGNAL(triggered()), this, SLOT(add_video()));
+    } else if(item->type == TYPE::VIDEO) {
+        selectedVideo = item;
+        QAction *loadVideo = new QAction(QIcon(""), tr("&Play video"), this);
+        loadVideo->setStatusTip(tr("Play video"));
+        menu.addAction(loadVideo);
+        connect(loadVideo, SIGNAL(triggered()), this, SLOT(play_video()));
+    }
+
+
+
+    QPoint pt(pos);
+    menu.exec( tree->mapToGlobal(pos) );
+}
+/**
+ * @brief MainWindow::add_video
+ * Prompts user with file browser to add video
+ * to selected project
+ */
+void MainWindow::add_video() {
+    QString dir = QFileDialog::getOpenFileName(this, tr("Choose video"),WORKSPACE,tr("*.avi;;*.mkv;;*.mov"));
```

Comment:
yes it does limit to those formats, the list should be extended. will extend to formats mentioned